lib/msf/core/payload/pingback.rb

```diff
@@ -0,0 +1,49 @@
+# -*- coding => binary -*-
+
+require 'msf/core'
+require 'msf/core/module/platform'
+require 'rex/text'
+
+#
+# This class provides methods for calculating, extracting, and parsing
+# unique ID values used by payloads.
+#
+module Msf::Payload::Pingback
+
+  #
+  # Instance methods
+  #
+    #
+  # Generates a URI with a given checksum and optionally with an embedded UUID if
+  # the desired length can accommodate it.
+  #
+  # @param mode [Symbol] The type of checksum to generate (:connect, :init_native, :init_python, :init_java)
+  # @param len [Integer] The length of the URI not including the leading slash, optionally nil for random
+  # @return [String] A URI with a leading slash that hashes to the checksum, with an optional UUID
+  #
+
+  # Generate a Payload UUID
+  def generate_pingback_uuid
+    self.pingback_uuid ||= SecureRandom.uuid()
+    print_status("PingbackUUID = #{self.pingback_uuid}")
+    if framework.db.active
+      print_status("Writing UUID #{datastore['PingbackUUID']} to database...")
+      Mdm::Payload.create!(name: datastore['PayloadUUIDName'],
```

Comment:
All Mdm code should live under `lib/msf/core/db_manager`. Interaction with `Mdm::Payload` should be possible via the `Msf::DBManager::Payload` module methods.